commit 26d074199730e215d35a62e437997693a9edb251
Author: Alexey Brodkin <alexey.brodkin@synopsys.com>
Date:   Wed Oct 31 18:25:47 2018 +0300

    devres: Align data[] to ARCH_KMALLOC_MINALIGN
    
    commit a66d972465d15b1d89281258805eb8b47d66bd36 upstream.
    
    Initially we bumped into problem with 32-bit aligned atomic64_t
    on ARC, see [1]. And then during quite lengthly discussion Peter Z.
    mentioned ARCH_KMALLOC_MINALIGN which IMHO makes perfect sense.
    If allocation is done by plain kmalloc() obtained buffer will be
    ARCH_KMALLOC_MINALIGN aligned and then why buffer obtained via
    devm_kmalloc() should have any other alignment?
    
    This way we at least get the same behavior for both types of
    allocation.
    
    [1] http://lists.infradead.org/pipermail/linux-snps-arc/2018-July/004009.html
    [2] http://lists.infradead.org/pipermail/linux-snps-arc/2018-July/004036.html
    
    Signed-off-by: Alexey Brodkin <abrodkin@synopsys.com>
    Cc: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
    Cc: Geert Uytterhoeven <geert@linux-m68k.org>
    Cc: David Laight <David.Laight@ACULAB.COM>
    Cc: Peter Zijlstra <peterz@infradead.org>
    Cc: Thomas Gleixner <tglx@linutronix.de>
    Cc: Vineet Gupta <vgupta@synopsys.com>
    Cc: Will Deacon <will.deacon@arm.com>
    Cc: Greg KH <greg@kroah.com>
    Cc: <stable@vger.kernel.org> # 4.8+
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/base/devres.c b/drivers/base/devres.c
index f98a097e73f2..d68b52cf9225 100644
--- a/drivers/base/devres.c
+++ b/drivers/base/devres.c
@@ -24,8 +24,14 @@ struct devres_node {
 
 struct devres {
 	struct devres_node		node;
-	/* -- 3 pointers */
-	unsigned long long		data[];	/* guarantee ull alignment */
+	/*
+	 * Some archs want to perform DMA into kmalloc caches
+	 * and need a guaranteed alignment larger than
+	 * the alignment of a 64-bit integer.
+	 * Thus we use ARCH_KMALLOC_MINALIGN here and get exactly the same
+	 * buffer alignment as if it was allocated by plain kmalloc().
+	 */
+	u8 __aligned(ARCH_KMALLOC_MINALIGN) data[];
 };
 
 struct devres_group {
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1000_26d074199730e215d35a62e437997693a9edb251.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1000_26d074199730e215d35a62e437997693a9edb251.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
